### PR TITLE
docs: Add c7i-flex instance-types reference

### LINKED ISSUE #6606: Missing c7i-flex instance family reference
### Description

**How can the docs be improved?**

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by jukie**
Fixes #6606

**Description**
Adds instance-type reference docs for c7i-flex
**How was this change tested?**
n/a
**Does this change impact docs?**
- [x] Yes, PR includes docs updates 
- [x] Yes, issue opened: #6606
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/preview/reference/instance-types.md [hunk 1]
```diff
@@ -4646,6 +4646,137 @@ below are the resources available with some assumptions and after the instance o
...
+ |karpenter.k8s.aws/instance-category|c|
+ |karpenter.k8s.aws/instance-cpu|2|
+ |karpenter.k8s.aws/instance-cpu-manufacturer|intel|
+ |karpenter.k8s.aws/instance-ebs-bandwidth|10000|
+ |karpenter.k8s.aws/instance-encryption-in-transit-supported|true|
+ |karpenter.k8s.aws/instance-family|c7i-flex|
+ |karpenter.k8s.aws/instance-generation|7|
+ |karpenter.k8s.aws/instance-hypervisor|nitro|
+ |karpenter.k8s.aws/instance-memory|4096|
+ |karpenter.k8s.aws/instance-network-bandwidth|781|
```
Comment (jukie) [inline website/content/en/preview/reference/instance-types.md:4663]: Where is this value pulled from? I just copied from c7i here

Comment (jukie) [reply website/content/en/preview/reference/instance-types.md]: I think these are correct now

**Comment by jukie** [comment]
Looks like this is autogenerated actually, will take a look at why c7i-flex isn't already here.

**Comment by jonathan-innis** [comment]
Closing since we merged #6815 with auto-gen.

**Pull request closed without merging**